### PR TITLE
FEATURE: TMA-320 Computed attributes distribution support in SDK and bricks

### PR DESCRIPTION
**Pull request by vhtien**

### CODE REVIEW

**Comment by vhtien** [comment]
PLS DO NOT MERGE

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/495/ : SUCCESS in 34s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/434/ : FAILURE in 26s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/171/ : SUCCESS in 1m 46s

**Comment by vhtien** [comment]
Tested code

```ruby
begin
  client = GoodData.connect <USERNAME>, <PASSWORD>, server: <SERVER>, verify_ssl: false
  bp = GoodData::Model::ProjectBlueprint.build("Xen's project from blueprint CA") do |project_builder|
    project_builder.add_date_dimension('created_on')

    project_builder.add_dataset('dataset.users', title: 'Users Dataset') do |schema_builder|
      schema_builder.add_anchor('attr.users.id', title: 'Users ID', folder: 'Users ID folder')
      schema_builder.add_label('label.users.id_label1', reference: 'attr.users.id')
      schema_builder.add_label('label.users.id_label2', reference: 'attr.users.id', default_label: true)
      schema_builder.add_attribute('attr.users.another_attr', title: 'Another attr')
      schema_builder.add_label('label.users.another_attr_label', reference: 'attr.users.another_attr')
      schema_builder.add_date('created_on')
      schema_builder.add_fact('fact.users.some_number')
    end
  end

  unless bp.valid?
    pp bp.validate
    fail "Origin blueprint is not valid"
  end

  project = client.create_project_from_blueprint(bp, auth_token: <TOKEN>, environment: 'TESTING')
  puts "Created project: #{project.pid}"

  metric = project.facts('fact.users.some_number').create_metric(title: 'Test')
  metric.save

  attribute = project.attributes('attr.users.another_attr')
  fail "is a CA" if attribute.computed_attribute?

  update = GoodData::Model::ProjectBlueprint.build('update') do |project_builder|
    project_builder.add_computed_attribute(
      'attr.comp.my_computed_attr',
      title: 'My computed attribute',
      metric: metric,
      attribute: attribute,
      buckets: [{ label: 'Xen', highest_value: 1000 }, { label: 'Ken', highest_value: 2000 }, { label: 'Kute', highest_value: 3000 }, { label: 'Olala' }]
    )
  end

  new_bp = bp.merge(update)

  unless new_bp.valid?
    pp new_bp.validate
    fail "New blueprint is not valid"
  end

  project.update_from_blueprint(new_bp)
  ca = project.attributes.find { |a| a.title == 'My computed attribute' }
  fail "CA is nil" unless ca
  fail "not a CA" unless ca.computed_attribute?
  puts project.computed_attributes.length

  ext_project = client.create_project_from_blueprint(project.blueprint(include_ca: false), auth_token: <TOKEN>, environment: 'TESTING')
  puts "Created ext project: #{ext_project.pid}"
  ext_ca = ext_project.attributes.find { |a| a.title == 'My computed attribute' }
  fail "fould CA in ext project" if ext_ca

  project.partial_md_export(metric.uri, project: ext_project)

  ext_project.update_from_blueprint(project.blueprint)
  ext_ca = ext_project.attributes.find { |a| a.title == 'My computed attribute' }
  fail "ext ca is nil" unless ext_ca
  fail "not a CA" unless ext_ca.computed_attribute?
  puts ext_project.computed_attributes.length
ensure
  project && project.delete
  ext_project && ext_project.delete
  client && client.disconnect
end
```

**Comment by vhtien** [comment]
bring tested script to another PR for comments: https://github.com/gooddata/gooddata-ruby/pull/875

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/498/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/437/ : FAILURE in 5m 27s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/174/ : SUCCESS in 1m 35s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/499/ : FAILURE in 9s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/438/ : FAILURE in 48m 01s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/175/ : SUCCESS in 1m 52s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/500/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/439/ : SUCCESS in 46m 08s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/176/ : SUCCESS in 1m 27s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/503/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/442/ : SUCCESS in 42m 53s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/179/ : SUCCESS in 1m 47s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/504/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/443/ : FAILURE in 45m 04s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/180/ : SUCCESS in 2m 07s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/505/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/444/ : SUCCESS in 43m 08s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/181/ : SUCCESS in 1m 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/512/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/451/ : SUCCESS in 43m 37s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/188/ : SUCCESS in 1m 47s

File: spec/integration/blueprint_with_ca_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require 'gooddata'
+
+describe 'Create project using GoodData client', :constraint => 'slow' do
+  before(:all) do
+    @client = ConnectionHelper.create_default_connection
+    @blueprint = GoodData::Model::ProjectBlueprint.build("My project from blueprint") do |project_builder|
+      project_builder.add_date_dimension('created_on')
```
Comment (panjan) [inline spec/integration/blueprint_with_ca_spec.rb]: In order to keep things simple, please create only metadata that we need for the test.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/515/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/454/ : FAILURE in 43m 16s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/191/ : SUCCESS in 2m 03s

**Comment by vhtien** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/516/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/455/ : SUCCESS in 43m 08s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/192/ : SUCCESS in 1m 34s

**Pull request merged**